src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,33 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.Ui;
+import duke.tasks.TaskList;
+
+/**
+ * Abstract class for all commands
+ */
+public abstract class Command {
+    private final boolean isExit;
+
+    public Command(boolean isExit) {
+        this.isExit = isExit;
+    }
+
+    /**
+     * Executes command
+     *
+     * @param tasks List of tasks
+     * @param ui User interface to print task
+     * @param storage File storage object
+     * @throws DukeException if exception encountered
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
```

Comment:
Change the method to abstract if you intent for the child classes to set their own logic.